File: src/main/scala/WordCountBatch.scala

import org.apache.flink.api.scala.{DataSet, ExecutionEnvironment, createTypeInformation}
import org.apache.flink.streaming.api.datastream.DataStream
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment

object WordCountBatch {


  def main(args: Array[String]): Unit = {

    // 执行环境
    val env : ExecutionEnvironment = ExecutionEnvironment.getExecutionEnvironment

    // source (file or stream)
    val txtStream : DataSet[String] = env.readTextFile("./input/words.txt")

    txtStream.flatMap(_.split(" "))
      .map((_, 1))
      .groupBy(0)
      .sum(1)
      .print()
//
//    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment()
//
//
//    val stream = env.socketTextStream("localhost", 18899)
//
//    val rstDataStream: DataStream[(Stream, Int)]= stream.flatMap(_.split(" "))
//      .filter(_.nonEmpty)
//      .map((_, 1))
//      .keyBy(0)
//      .sum(1)
//
//    rstDataStream.print()


  }
}

File: src/main/java/myflink/wordcount/WordCountBatch.java
package myflink.wordcount;

import org.apache.flink.api.common.typeinfo.Types;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.AggregateOperator;
import org.apache.flink.api.java.operators.DataSource;
import org.apache.flink.api.java.operators.FlatMapOperator;
import org.apache.flink.api.java.operators.UnsortedGrouping;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.util.Collector;


public class WordCountBatch {

  public static void main(String[] args) throws Exception {

      // 1. 创建执行环境

      ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();

      // 2. 从本地文件读取数据
      DataSource<String> stringDataSource = env.readTextFile("input/words.txt");

      // 3. 将每行数据 分词
      FlatMapOperator<String, Tuple2<String, Long>> wordAndOneTuple = stringDataSource.flatMap((String line, Collector<Tuple2<String, Long>> collector) -> {
          String[] words = line.split(" ");
          for (String word : words) {
              collector.collect(new Tuple2(word, 1L));
          }
      }).returns(Types.TUPLE(Types.STRING, Types.LONG));


      // 4. 按 word 分组
      UnsortedGrouping<Tuple2<String, Long>> wordAndGroup = wordAndOneTuple.groupBy(0);

      // 5. 分组内进行聚合统计
      AggregateOperator<Tuple2<String, Long>> aggResult = wordAndGroup.sum(1);



      aggResult.print();



//      env.execute();
  }
}